src/main/java/org/example/controllers/ApartmentController.java

```diff
@@ -0,0 +1,54 @@
+package org.example.controllers;
+
+import org.example.dto.ApartmentDTO;
+import org.example.service.ApartmentService;
+import org.example.service.RealtyAgentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Controller
+public class ApartmentController {
+
+    @Autowired
+    protected ApartmentService apartmentService;
+
+    @Autowired
+    protected RealtyAgentService realtyAgentService;
+
+    @GetMapping("/apartmentlist")
+    public String showApartments(@RequestParam("agent.id") int agentId, ModelMap modelMap) {
```

Comment:
It's better to avoid using 'complex' parameters, like `agent.id`. Replace this and others to something like `afentId`